### ISSUE TITLE
Training Time For Multinode is similar to training time for single node

### ISSUE THREAD

**Issue by thakursc1** [huggingface/accelerate#91]
@sgugger  As per our discussion on #37
Here is my method of launch and results
I am using AML Per process launcher shown here (option 1): 
https://azure.github.io/azureml-cheatsheets/ja/docs/cheatsheets/python/v1/distributed-training/
To launch a distributed training job with multiple process per nodes. 

I am using traditional run_mlm.py (without trainer from transformers repository [here](https://github.com/huggingface/transformers/blob/master/examples/pytorch/language-modeling/run_mlm_no_trainer.py) )

When I observe the tqdm log from when running for 1 node vs running with 8 nodes:
I observe the following:

8 Node log : 

INFO:__main__:***** Running training *****
INFO:__main__:  Num examples = 4842767
INFO:__main__:  Num Epochs = 10
INFO:__main__:  Instantaneous batch size per device = 64
INFO:__main__:  Total train batch size (w. parallel, distributed & accumulation) = 2048
INFO:__main__:  Gradient Accumulation steps = 1
INFO:__main__:  Total optimization steps = 23650

  0%|          | 20/23650 [01:20<26:05:59,  3.98s/it]
  0%|          | 21/23650 [01:24<26:04:43,  3.97s/it]
  0%|          | 22/23650 [01:28<26:12:45,  3.99s/it]

1Node Log: 
INFO:__main__:***** Running training *****
INFO:__main__:  Num examples = 4842767
INFO:__main__:  Num Epochs = 10
INFO:__main__:  Instantaneous batch size per device = 64
INFO:__main__:  Total train batch size (w. parallel, distributed & accumulation) = 256
INFO:__main__:  Gradient Accumulation steps = 1
INFO:__main__:  Total optimization steps = 189180

  0%|          | 7/189180 [00:04<27:49:11,  1.89it/s]
  0%|          | 8/189180 [00:04<27:27:56,  1.91it/s]

As you can see the iteration time per step for single node is about 8 times higher than 8 node.
You can see training time estimated for one node is lesser than 8 nodes.

**Comment by sgugger** [comment]
Thanks for opening a new issue. I am not sure how I can help however: `Accelerate` does not provide any optimization on top of PyTorch, it's just syntax sugar to make it easy to launch your distributed training. Behind the scenes, the model is a regular PyTorch DDP model as recommended by the documentation you linked. It looks more like an issue with the cloud provider (there is some sort of communication bottleneck between the nodes).

**Comment by thakursc1** [comment]
@sgugger  Can you confirm is accelerate is running as expected from these logs ? 

Also i observed the backend being used is gloo. Do you think this is causing the issue ?

**Comment by sgugger** [comment]
It looks okay to me! Accelerate initializes the backend with nccl if it's not already initialized, not sure if this has an impact in your case.

**Comment by thakursc1** [comment]
Figured it out, seems like AML has a communication bottleneck indeed. To get any relevant gain due to node scaling requires RDMA enabled compute cluster and Infiniband enabled cluster.
